hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/OzoneClientProducer.java

```diff
@@ -106,7 +107,7 @@ public S3Auth getSignature() {
       // For any other critical errors during object creation throw Internal
       // error.
       LOG.debug("Error during Client Creation: ", e);
-      throw wrapOS3Exception(INTERNAL_ERROR);
+      throw wrapOS3Exception(S3ErrorTable.getInternalError(e));
```

Comment:
It seems improving `wrapOS3Exception` is enough for getting friendly message for error correctable by user.  I think we should skip this part of the change (and the one in `S3ErrorTable`).

```suggestion
      throw wrapOS3Exception(INTERNAL_ERROR);
```